lib/generate_puppetfile/bin.rb

```diff
@@ -6,15 +6,21 @@
 require 'json'
 require 'mkmf'
 require 'colorize'
+require 'uri'
 
 module GeneratePuppetfile
   # Internal: The Bin class contains the logic for calling generate_puppetfile at the command line
   class Bin
-    Module_Regex = Regexp.new("mod ['\"]([a-z0-9_]+\/[a-z0-9_]+)['\"](,\\s+['\"](\\d+\.\\d+\.\\d+)['\"])?\\s+$", Regexp::IGNORECASE)
-    @options = {}         # Options hash
-    @workspace = nil      # Working directory for module download and inspection
-    @module_data = {}     # key: modulename, value: version number
-    @download_errors = '' # A list of errors encountered while downloading modules. Should remain empty.
+    Module_Regex        = %r{^\s*mod ['\"]([a-z0-9_]+\/[a-z0-9_]+)['\"](,\s+['\"](\d+\.\d+\.\d+)['\"])?\s*$}
+    Repository_Regex    = %r{^\s*mod\s+['\"](\w+)['\"]\s*,\s*$}
+    Location_Only_Regex = %r{^\s+:git\s+=>\s+['\"](\S+)['\"]\s*$}
+    Location_Plus_Regex = %r{^\s+:git\s+=>\s+['\"](\S+)['\"]\s*,\s*$}
+    Type_ID_Regex       = %r{^\s+:(\w+)\s+=>\s+['\"](\S+)['\"]\s*$}
+    Forge_Regex         = %r{^forge}
+    Blanks_Regex        = %r{^\s*$}
+    Comments_Regex      = %r{^#}
+    Skipall_Regex       = %r{^forge|^\s*$|^#}
```

Comment:
I think the comment regex (in `Skipall_Regex`) should allow for spaces beforehand: `%r{^forge|^\s*$|^\s*#}`

Currently, the following would not be accepted:
```
mod "foo-bar",
    # we need a specific commit until the forge is updated
    :git => 'git://github.com/foo/puppet-bar',
    :ref => '4242d00d'
```